test cases/common/238 fstrings/meson.build

```diff
@@ -0,0 +1,7 @@
+project('meson-test', 'c')
+
+n = 10
+m = 'bar'
+s = f'test @n@ string (@@n@@): @m@'
+
+assert(s == 'test 10 string (@10@): bar', 'Incorrect string formatting')
```

Comment:
This seems a bit strange. Typically doubling of the quote character prints a single quote character and nothing else. For example in Python:

```python
>>> x = 3
>>> f'{{x}}'
'{x}'
```

Reply:
I agree that's wrong, but as per previous comments in this PR, it is already the case for .format() too.

Reply:
That's true but this behavior is consistent with `.format()`.

Reply:
If we choose to adjust this behavior I'd say we should update both fstrings and `.format()`. I'm not sure if that's an acceptable amount of breakage though.

Reply:
Doing it properly is probably harder than a simple regex if I'm not mistaken. Let's not block this PR on that, if it ever becomes an issue we should fix both fstring and .format() anyway.